widgetastic_manageiq/__init__.py

```diff
@@ -4356,3 +4356,107 @@ def fill(self, value):
             return False
         self._all_item_elements[value or self.selected_item].click()
         return True
+
+
+class MigrationPlanRequestDetailsList(Widget):
+    """Represents the list of Migration Plan Request Details."""
+
+    ROOT = ParametrizedLocator('.//div[contains(@class,{@list_class|quote})]')
+    ITEM_LOCATOR = ParametrizedLocator('./div[contains(@class,"list-group-item")]')
+    # ITEM_TEXT_LOCATOR helps locate name of the Migration plan. ITEM_LOCATOR.text does not suffice.
+    # Also, ITEM_LOCATOR does not yield element which can be clicked to navigate to details page.
+    ITEM_TEXT_LOCATOR = './/div[contains(@class,"list-group-item-heading")]'
+    ITEM_MESSAGE_LOCATOR = './/div[./button/span[contains(@class,"pficon-info")]]'
+    ITEM_TIMER_LOCATOR = './/div[./span[contains(@class,"fa-clock-o")]]'
+    ITEM_ADDITIONAL_INFO_BUTTON_LOCATOR = './/div/button[./span[contains(@class,"pficon-info")]]'
+    ITEM_PROGRESS_DESCRIPTION_LOCATOR = './/div[contains(@class,"progress-description ")]'
+    ITEM_DOWNLOAD_LOG_BUTTON_LOCATOR = './/div[./a][contains(@class,"list-view-pf-actions")]'
+    ITEM_IS_ERRORED_LOCATOR = './/div[contains(@class,"pficon-error-circle-o")]'
+    ITEM_PROGRESS_SPINNER_LOCATOR = './/div[contains(@class,"spinner")]'
+    ITEM_IS_SUCCESSFUL_LOCATOR = './/div[contains(@class,"pficon-ok")]'
+
+    def __init__(self, parent, list_class, logger=None):
+        Widget.__init__(self, parent, logger=logger)
+        self.list_class = list_class
+
+    @property
+    def all_items(self):
+        return [self.browser.text(item) for item in self.browser.elements(self.ITEM_TEXT_LOCATOR)]
+
+    def _get_vm_element(self, vm_name):
+        for item in self.browser.elements(self.ITEM_LOCATOR):
+            el = self.browser.element(".//*[@class='list-group-item-heading']", parent=item)
+            if self.browser.text(el) == vm_name:
+                return item
+        raise ItemNotFound("VM: {} not found".format(vm_name))
+
+    def read(self):
+        return self.all_items
+
+    def fill(self, value):
```

Comment:
**REQUIRED**
In that case do not define `fill()` at all.

Reply:
Please review https://github.com/ManageIQ/integration_tests/pull/7343/files for this widget, once that is merged, I will rebase this PR.